Specifications/Applications/for_ApplicationArtifactIdentifier/given/identifiers_with_artifacts_of_different_generations.cs

```diff
@@ -1,32 +0,0 @@
-using Dolittle.Artifacts;
-using Machine.Specifications;
-using Moq;
-
-namespace Dolittle.Applications.Specs.for_ApplicationArtifactIdentifier.given
-{
-    public class identifiers_with_artifacts_of_different_generations
```

Comment:
Reasoning for removal is that ApplicationArtufactIdentifier should now know how ArtifactGeneration behaves relative to the Artifact. The classes that derives from IArtifact should define how equality and HashCode is defined  and how it should function.

Unless we want the IArtifact property of ApplicationArtifactIdentifier to be a concrete type, Artifact, then we could do these kinds of tests